src/composables/websocket.ts

```diff
@@ -0,0 +1,155 @@
+import { ref } from '@vue/composition-api';
+import camelcaseKeysDeep from 'camelcase-keys-deep';
+import { IDefaultComposableOptions, INetwork, WebSocketChannelTypes } from '../types';
+import { createNetworkWatcher } from './composablesHelpers';
+import { useGetter } from './vuex';
+
+const { onNetworkChange } = createNetworkWatcher();
+
+const wsClient = ref<WebSocket>();
+const isConnected = ref(false);
+const subscribers = ref<Record<WebSocketChannelTypes, Array<any>>>({
+  Transactions: [],
+  MicroBlocks: [],
+  KeyBlocks: [],
+});
+const channels = ref<Record<WebSocketChannelTypes, boolean>>({
+  Transactions: false,
+  MicroBlocks: false,
+  KeyBlocks: false,
+});
+
+export function useWebsocket({ store }: IDefaultComposableOptions) {
+  const activeNetwork = useGetter<INetwork>('activeNetwork');
+
+  /**
+   * This function is used to subscribe to a specific WebSocket channel.
+   * @param {WebSocketChannelTypes} channelName - The name of the channel to subscribe to
+   */
+  function subscribeToWebsocketChannel(channelName: WebSocketChannelTypes) {
+    if (
+      !channelName
+      || channels.value[channelName]
+      || !isConnected.value
+      || !wsClient.value
+    ) return;
+
+    wsClient.value.send((JSON.stringify({
+      op: 'Subscribe',
+      payload: channelName,
+    })));
```

Comment:
I suggest to create an interface `IMiddlewareWebSocketSubscriptionMessage` in our types directory and comment it with the reference link to https://github.com/aeternity/ae_mdw#websocket-interface